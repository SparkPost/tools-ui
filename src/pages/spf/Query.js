/* eslint-disable no-console */

import React, { Component } from 'react';
import { withRouter } from 'react-router';
import classNames from 'classnames';
import { INTRO_TEXT } from './constants';
import { LinkButton } from 'components/button/Button';


class Query extends Component {
  constructor(props) {
    super(props);
    this.state = {
      domain: '',
      domainError: false
    };
  }

  validateDomain() {
    // does this sorta kinda look like a domain
    return this.state.domain && this.state.domain.indexOf('.') > 0;
  }

  componentDidMount() {
    this.props.router.setRouteLeaveHook(this.props.route, () => {
      const validDomain = this.validateDomain(this.state.domain);
      this.setState({domainError: !validDomain});
      return validDomain;
    });
  }

  renderPanel() {
    return (
      <div className='panel panel--accent text--left'>
        <div className='panel__body'>
          <h4>Inspect a Domain</h4>
          <p className='paddingBottom--md text--muted'>Enter a domain to inspect, e.g., sparkpost.com.</p>
          { this.renderForm() }
        </div>
      </div>
    );
  }

  renderForm() {
    return (
      <form>
        <div className="flex">

          <div className={classNames('col-xs-8', {'has-error': this.state.domainError })}>
            <input className='input__text input--full' type='text' onChange={(e) => this.setState({domain: e.target.value}) } placeholder='mydomain.com' />
            <span className="input__error">Please enter a valid domain.</span>
          </div>

          <div className='col-xs-4'>
            <LinkButton type='orange' fullWidth={true} to={ `/spf/results/${this.state.domain}` }>View Results</LinkButton>
          </div>

        </div>
      </form>
    );
  }

  render() {
    return (
      <div className='flex center-xs'>
        <div className='col-xs-12 col-md-10 col-lg-7'>
          <h1>SPF Inspector</h1>
          <p className='marginBottom--lg text--muted'>{INTRO_TEXT}</p>
          { this.renderPanel() }
        </div>
      </div>
    );
  }
}

export default withRouter(Query);
