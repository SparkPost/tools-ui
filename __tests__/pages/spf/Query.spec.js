import React from 'react';
import renderer from 'react-test-renderer';
import Query from 'pages/spf/Query';

describe('Query Snapshots', () => {
  test('baseline snapshot', () => {
    expect(renderer.create(<Query></Query>)).toMatchSnapshot();
  });
});
