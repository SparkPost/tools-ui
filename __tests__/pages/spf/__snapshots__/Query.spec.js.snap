exports[`Query Snapshots baseline snapshot 1`] = `
<div
  className="flex center-xs">
  <div
    className="col-xs-12 col-md-10 col-lg-7">
    <h1>
      SPF Inspector
    </h1>
    <p
      className="marginBottom--lg text--muted">
      Visualize your SPF records. Identify which servers are authorized to send on behalf of a domain.
    </p>
    <div
      className="panel panel--accent text--left">
      <div
        className="panel__body">
        <h4>
          Inspect a Domain
        </h4>
        <p
          className="paddingBottom--md text--muted">
          Enter a domain to inspect, e.g., sparkpost.com.
        </p>
        <form
          onSubmit={[Function]}>
          <div
            className="flex">
            <div
              className="col-xs-12 col-md-8">
              <div
                className="input__group">
                <input
                  className="input__text input--full"
                  onChange={[Function]}
                  placeholder="mydomain.com"
                  type="text" />
                <div
                  className="input__error">
                  Please enter a valid domain.
                </div>
              </div>
            </div>
            <div
              className="col-xs-12 col-md-4">
              <button
                className="button button--orange button--full"
                type="submit">
                View Results
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
